resources/nomad_client.go

```diff
@@ -84,16 +86,47 @@ func (nc NomadClient) Scale(newCount int, vc *VaultClient) error {
 	*tg.Count = newCount
 	*job.VaultToken = vc.GetVaultToken()
 
-	logging.Info("Old nomad: %d. Desired nomad: %d", oldCount, newCount)
-
 	_, _, err = nc.client.nomad.Jobs().Register(job, &nomad.WriteOptions{})
 	if err != nil {
 		logging.Error(err.Error())
 		return err
 	}
 
 	count, _ := nc.GetTaskGroupCount()
-	logging.Info("Old nomad: %d. New nomad: %d", oldCount, count)
+	logging.Info("[scaling log] Nomad job: %s. Old: %d. New: %d", nc.JobName, oldCount, count)
+
+	return nil
+}
+
+// RestartNomadAlloc - restart any Nomad allocation of the associated job in NomadClient
+func (nc NomadClient) RestartNomadAlloc() error {
+	allocs, _, err := nc.client.nomad.Jobs().Allocations(nc.JobName, true, nil)
+	if err != nil {
+		return err
+	}
+	if len(allocs) > 0 {
+		allocID := ""
+		for i := range allocs {
+			if allocs[i].ClientStatus == "running" {
+				allocID = allocs[i].ID
+				break
+			}
+		}
+		if allocID == "" {
+			return fmt.Errorf("no running allocations to restart")
+		}
+		alloc, _, err := nc.client.nomad.Allocations().Info(allocID, nil)
+		if err != nil {
+			return err
+		}
+		logging.Info("[restart log] stopping %s", alloc.ID)
+		_, err = nc.client.nomad.Allocations().Stop(alloc, nil)
+		if err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("no allocations to restart")
+	}
 	return nil
```

Comment:
i think this is unreachable code that could be removed or just not have the else statement 👍

Reply:
It's reachable if no errors along the way

Reply:
I see, i got confused as there was a return in the else block. Would it be neater if you do something like 

```
if len(allocs) <  {
    return fmt.Errorf("no allocations to restart")
}

// rest of code
```

Reply:
Good point 👍, fixed in 926bb901308c9af0e87a694a4a4dc4a499fff061

Reply:
Thnx! ill leave the lgtm to @guangie88. But its nice to know its still being used 😆

Reply:
Actually this is a good catch, helps to maintain the early return error first, then only return "success" value at the end. :+1: